test/smoke/src/areas/positron/welcome/welcome.test.ts

```diff
@@ -0,0 +1,189 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (C) 2024 Posit Software, PBC. All rights reserved.
+ *  Licensed under the Elastic License 2.0. See LICENSE.txt for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+
+import { expect } from '@playwright/test';
+import { Application, Logger, PositronPythonFixtures, PositronRFixtures } from '../../../../../automation';
+import { installAllHandlers } from '../../../utils';
+
+/*
+ *  Welcome test cases.
+ */
+
+export function setup(logger: Logger) {
+	describe('Welcome Page', () => {
+		installAllHandlers(logger);
+
+		beforeEach(async function () {
+			const app = this.app as Application;
+
+			await app.workbench.quickaccess.runCommand('Help: Welcome');
+			app.workbench.editors.waitForActiveEditor('Welcome');
+		});
+
+		afterEach(async function () {
+			const app = this.app as Application;
+			await app.workbench.quickaccess.runCommand('View: Close All Editors');
+		});
+
+		it('Verify Welcome page header and footer', async function () {
+			const app = this.app as Application;
+			const logoLocator = '.product-logo';
+			const titleLocator = '.gettingStartedCategoriesContainer div.header div .positron';
+			const footerLocator = '.gettingStartedCategoriesContainer div.footer';
+
+			const footer = app.code.driver.getLocator(footerLocator);
+
+			const logo = app.code.driver.getLocator(logoLocator);
+
+			await expect(logo).toBeVisible();
+			const logoBox = await (await logo.elementHandle())?.boundingBox();
+
+			expect(logoBox?.height).toBeGreaterThan(0);
+			expect(logoBox?.width).toBeGreaterThan(0);
+
+			const title = app.code.driver.getLocator(titleLocator);
+
+			// product name in release is 'Positron' and in dev is 'Positron Dev'
+			await expect(title).toHaveText([/(Positron)|(Positron Dev)/, 'an IDE for data science']);
+
+			await expect(footer).toHaveText('Show welcome page on startup');
+		});
+
+		it('Verify Welcome page content', async function () {
+			const app = this.app as Application;
+
+			await expect(app.workbench.positronWelcome.startTitle).toHaveText('Start');
+
+			await expect(app.workbench.positronWelcome.startButtons).toHaveCount(4);
+			await expect(app.workbench.positronWelcome.startButtons).toHaveText(['New Notebook', 'New File', 'New Console', 'New Project']);
+
+			await expect(app.workbench.positronWelcome.helpTitle).toHaveText('Help');
+
+			await expect(app.workbench.positronWelcome.helpLinks).toHaveCount(3);
+			await expect(app.workbench.positronWelcome.helpLinks).toHaveText(['Positron Documentation', 'Positron Community', 'Report a bug']);
+
+			await expect(app.workbench.positronWelcome.openTitle).toHaveText('Open');
+
+			await expect(app.workbench.positronWelcome.openButtons).toHaveCount(3);
```

Comment:
This needs a change since Mac is the only platform that doesn't have the `Open File...` action.